pkg/nmi/server/server.go

```diff
@@ -155,6 +158,21 @@ func (s *Server) hostHandler(logger *log.Entry, w http.ResponseWriter, r *http.R
 		http.Error(w, msg, http.StatusForbidden)
 		return
 	}
+
+	// filter out if we are in namesoaced mode
+	filterPodIdentities := []aadpodid.AzureIdentity{}
+	for _, val := range *(podIDs) {
+		if s.IsNamespaced || aadpodid.IsNamespacedIdenity(&val) { // namespaced mode
```

Comment:
optional: can reduce the branching by using if (condition) { assignment; continue} pattern